### DISCUSSION TITLE
Validation failed when PR already exists

### DISCUSSION THREAD

**Discussion by joakimgunst** [crowdin/github-action#193] [Q&A] [answered]
When running the same job twice, I get the following error:

```
...
CHECK IF ISSET SAME PULL REQUEST
CREATE PULL REQUEST
FAILED TO CREATE PULL REQUEST
RESPONSE: {
  "message": "Validation Failed",
  "errors": [
    {
      "resource": "PullRequest",
      "code": "custom",
      "message": "A pull request already exists for joakimgunst:l10n_crowdin_translations."
    }
  ],
  "documentation_url": "https://docs.github.com/rest/pulls/pulls#create-a-pull-request"
}
...
```

The new changes are force-pushed to the localization branch so the PR is updated correctly, but the job fails. Is there a way to configure the action so that this doesn't happen?

Repo: https://github.com/joakimgunst/crowdin-test
Workflow: https://github.com/joakimgunst/crowdin-test/blob/main/.github/workflows/crowdin.yml
Failed job: https://github.com/joakimgunst/crowdin-test/actions/runs/6156831538/job/16706366420

**Comment by joakimgunst** [comment]
I suppose one option is to disable the PR creation in the job and open the PR manually, but ideally I would like a PR to be created (only) if none is currently open.

**Comment by andrii-bodnar** [comment]
Hi @joakimgunst,

It's strange because this action has the check to see if the same pull request exists in the repo - https://github.com/crowdin/github-action/blob/master/entrypoint.sh#L95-L113

Could you please try to run the workflow in debug mode?

**Comment by joakimgunst** [comment]
@andrii-bodnar Here is a debug run: https://github.com/joakimgunst/crowdin-test/actions/runs/6157076960/job/16709678134

**Comment by joakimgunst** [comment]
I created this script as a test, and it prints `CREATE PULL REQUEST`, so maybe there is a bug?

```sh
#!/bin/sh

PULL_REQUESTS='l10n_crowdin_translations
l10n_main
renovate/configure'

BRANCH='l10n_crowdin_translations'

if echo "$PULL_REQUESTS " | grep -q "$BRANCH "; then
  echo "PULL REQUEST ALREADY EXIST"
else
  echo "CREATE PULL REQUEST"
fi
```

**Comment by joakimgunst** [comment]
The pull request names are separated by newlines, and the extra whitespace after `$BRANCH` therefore doesn't match.

**Comment by joakimgunst** [comment]
I think if `jq` in this case outputs a newline-separated string of PR names, you might need e.g. `"$PULL_REQUESTS" | grep -xq "$BRANCH"` which matches only the whole line:

> -x, --line-regexp
> Select only those matches that exactly match the whole line.  For a regular expression pattern, this is like parenthesizing the pattern and then surrounding it with ^ and $.

**Comment by joakimgunst** [comment]
Or using [`join`](https://jqlang.github.io/jq/manual/#join) before the grep.

**Comment by andrii-bodnar** [comment]
@joakimgunst thanks a lot for the details provided!

I just prepared a possible [fix](https://github.com/crowdin/github-action/commit/b0270b7fb2079c9421b32e30c0d88d9c06d59330), could you please check it by referencing the commit hash as the Action version?

```yml
uses: crowdin/github-action@b0270b7fb2079c9421b32e30c0d88d9c06d59330
```

**Comment by joakimgunst** [comment]
It seems to work: https://github.com/joakimgunst/crowdin-test/actions/runs/6158980256/job/16712861357

**Comment by andrii-bodnar** [comment, marked as answer]
@joakimgunst glad to hear that! 

So I'm going to merge this fix and it will be available in the next Action release.

Thanks for the help with the debugging! 🚀

**Comment by joakimgunst** [comment]
Thanks for the quick response! 🙏
